docs/user-manual/graphics/shaders/index.md

```diff
@@ -0,0 +1,199 @@
+---
+title: Shaders
+sidebar_position: 6
+---
+
+When you import your 3D models into PlayCanvas by default they will use our [Physical Material][1]. This is a versatile material type that can cover a lot of your rendering needs.
+
+However, you will often want to perform special effects or special cases for your materials. To do this you will need to write a custom shader. In this cases, you need to use `ShaderMaterial`.
+
+To create an instance of `ShaderMaterial`, these are the steps:
+
+Create a description of your shader.
+
+``` javascript
+const shaderDesc = {
+        uniqueName: 'MyShader',
+        shaderLanguage: pc.SHADERLANGUAGE_GLSL,
+        vertexCode: `
+            // write your vertex shader source code
+        `,
+        fragmentCode: `
+            // write your fragment shader source code
+        `,
+        attributes: {
+            aPosition: pc.SEMANTIC_POSITION,
+            aUv0: pc.SEMANTIC_TEXCOORD0
+        }
+    };
+
+```
+
+and then create instances of your material, which you can use for rendering:
+
+``` javascript
+const material = new pc.ShaderMaterial(shaderDesc);
+```
+
+The shader source code can be written in GLSL if you're targeting the WebGL2 or WebGPU platforms, or in WGSL if you're targeting WebGPU only.
+
+:::note
+
+If you write a GLSL shader, it is directly supported by the WebGL2 platform. However, on the WebGPU platform, GLSL shaders require transpilation to WGSL using WASM transpilers. To avoid this transpilation step and achieve native performance related to shader compilation, and avoid additional download of WASM files, you might want to consider writing an equivalent shader in WGSL for the WebGPU platform, which is supported directly.
```

Comment:
```suggestion
If you write a GLSL shader, it is directly supported by the WebGL2 platform. However, on the WebGPU platform, GLSL shaders require transpilation to WGSL using a WASM transpiler. To avoid this transpilation step and achieve native performance related to shader compilation, and avoid additional download of WASM files, you might want to consider writing an equivalent shader in WGSL for the WebGPU platform, which is supported directly.
```